server/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -219,19 +220,23 @@ private DivTag renderQuestionSummary(AnswerData data, Messages messages, long ap
     LinkElement editElement =
         new LinkElement()
             .setStyles("bottom-0", "right-0", "text-blue-600", StyleUtils.hover("text-blue-700"));
+
+    QuestionDefinition questionDefinition = data.questionDefinition();
+    Optional<String> questionName = Optional.of(questionDefinition.getName());
+    Optional<String> questionType = Optional.of(questionDefinition.getQuestionType().toString());
     if (data.isAnswered()) {
       editElement
           .setHref(
               routes.ApplicantProgramBlocksController.review(
-                      applicantId, data.programId(), data.blockId())
+                      applicantId, data.programId(), data.blockId(), questionName, questionType)
                   .url())
           .setText(messages.at(MessageKey.LINK_EDIT.getKeyName()))
           .setIcon(Icons.EDIT, LinkElement.IconPosition.START);
     } else {
       editElement
           .setHref(
               routes.ApplicantProgramBlocksController.edit(
-                      applicantId, data.programId(), data.blockId())
+                      applicantId, data.programId(), data.blockId(), questionName, questionType)
```

Comment:
Are the links being urlencoded? Question name could have special characters in it.

Reply:
Good question, I'm not sure. I'll look into it.

Reply:
Looks like they are url encoded by the Java Play reverse routing. I tried out the admin id `special!` and the url generated was `http://localhost:9000/applicants/264/programs/41/blocks/1/edit?questionName=special%21`. Autofocus worked when clicking through to edit that question.